charmbenchmark/__init__.py

```diff
@@ -1,143 +1,11 @@
-                    })
-
-            # Check the relation data
-            config = {}
-            for key in self.required_keys:
-                val = relation_get(key)
-                if val is not None:
-                    config[key] = val
-                else:
-                    # We don't have all of the required keys
-                    config = {}
-                    break
-
-            if len(config):
-                with open('/etc/benchmark.conf', 'w') as f:
-                    for key, val in config.items():
-                        f.write("%s=%s\n" % (key, val))
-
-    @staticmethod
-    def set_data(value):
-        """
-        Set the key:value to be passed to ``action_set``
-
-        :param list value: A hash containing the key and value
-        """
-        try:
-            action_set(value)
-            return True
-        except OSError:
-            return False
-
-    @staticmethod
-    def set_composite_score(value, units='', direction=None):
-        """
-        Set the composite score for a benchmark run. This is a single number
-        representative of the benchmark results. This could be the most
-        important metric, or an amalgamation of metric scores.
-        """
-        Benchmark.set_data({'meta.composite.value': value})
-
-        # The GUI expects this key, even if it's empty
-        Benchmark.set_data({'meta.composite.units': units})
-
-        if direction:
-            Benchmark.set_data({'meta.composite.direction': direction})
-        return True
-
-    @staticmethod
-    def set_meta(key, value):
-        Benchmark.set_data({'meta.%s.value' % key: value})
-
-    @staticmethod
-    def start():
-        """
-        If the cabs-collector charm is installed, take a snapshot
-        of the current profile data.
-        """
-        # Do profile data collection immediately on this unit
-        if os.path.exists(COLLECT_PROFILE_DATA):
-            subprocess.check_output([COLLECT_PROFILE_DATA])
-
-        # Tell the benchmark-gui charm the action_id via the benchmark
-        # relation. Benchmark-gui will pass the action_id to all
-        # collector charms in the environment (via the collector relation),
-        # triggering profile data collection on each.
-        for rid in relation_ids('benchmark'):
-            relation_set(relation_id=rid, relation_settings={
-                'action_id': os.environ.get('JUJU_ACTION_UUID')
-            })
-
-        return Benchmark.set_data({
-            'meta.start': time.strftime('%Y-%m-%dT%H:%M:%SZ')
-        })
-
-    @staticmethod
-    def finish():
-        return Benchmark.set_data({
-            'meta.stop': time.strftime('%Y-%m-%dT%H:%M:%SZ')
-        })
+        warnings.warn("The charm-benchmark library has been renamed charms.benchmark. Please update your code accordingly or report a bug with the upstream project.", DeprecationWarning)
+        charms.benchmark.Benchmark.__init__(self, benchmarks)
```

Comment:
it would be better to use `super()` here instead of `charms.benchmark.Benchmark.__init__`

ie:

```python
    def __init__(self, benchmarks=None):
        warnings.warn("The charm-benchmark library has been renamed charms.benchmark. Please update your code accordingly or report a bug with the upstream project.", DeprecationWarning)
        super(ChildB, self).__init__(benchmarks)
```